crm_claim_code/models/crm_claim.py

```diff
@@ -29,3 +30,12 @@ def copy(self, default=None):
             default = {}
         default['code'] = self.env['ir.sequence'].get('crm.claim')
         return super(CrmClaim, self).copy(default)
+
+    @api.multi
+    @api.depends('code', 'name')
```

Comment:
@api.depends is for computed fields, not for this one.

You should also call super and prefix the code instead hardcoding it for allowing other personalizations.